multus_sriov_cni.sh

```diff
@@ -0,0 +1,247 @@
+    git clone $SRIOV_CNI_REPO $WORKSPACE/sriov-cni
+    pushd $WORKSPACE/sriov-cni
+    git checkout $SRIOV_CNI_BRANCH
+    git log -p -1 > $ARTIFACTS/sriov-cni.txt
+    popd
+
+    echo "Download $PLUGINS_REPO"
+    git clone $PLUGINS_REPO $WORKSPACE/plugins
+    pushd $WORKSPACE/plugins
+    git checkout $PLUGINS_BRANCH
+    git log -p -1 > $ARTIFACTS/plugins.txt
+    ./build_linux.sh
+    popd
+
+    echo "Download $SRIOV_NETWORK_DEVICE_PLUGIN_REPO"
+    git clone $SRIOV_NETWORK_DEVICE_PLUGIN_REPO $WORKSPACE/sriov-network-device-plugin
+    pushd $WORKSPACE/sriov-network-device-plugin
+    git checkout $SRIOV_NETWORK_DEVICE_PLUGIN_BRANCH
+    git log -p -1 > $ARTIFACTS/sriov-network-device-plugin.txt
+    make build
+    popd
+    mkdir -p /etc/pcidp/
+    cat > /etc/pcidp/config.json <<EOF
+{
+    "resourceList": [{
+        "resourceName": "sriov",
+        "selectors": {
+                "vendors": ["15b3"],
+                "devices": ["1018"],
+                "drivers": ["mlx5_core"]
+            }
+    }
+    ]
+}
+EOF
+    cp /etc/pcidp/config.json $ARTIFACTS
+    return 0
+}
+
+
+function pod_create {
+    cat > /tmp/sriov_pod.yaml <<EOF
+apiVersion: v1
+kind: Pod
+metadata:
+  name: mofed-test-pod-1
+  annotations:
+    k8s.v1.cni.cncf.io/networks: sriov-net1
+spec:
+  restartPolicy: OnFailure
+  containers:
+    - image: mellanox/mlnx_ofed_linux-4.4-1.0.0.0-centos7.4
+      name: mofed-test-ctr
+      imagePullPolicy: IfNotPresent
+      securityContext:
+        capabilities:
+          add: ["IPC_LOCK"]
+      resources:
+        limits:
+          intel.com/sriov: 1
+      command:
+        - sh
+        - -c
+        - |
+          ls -l /dev/infiniband /sys/class/net
+          sleep 1000000
+EOF
+    cp /tmp/sriov_pod.yaml $ARTIFACTS
+    kubectl delete -f /tmp/sriov_pod.yaml
+    kubectl create -f /tmp/sriov_pod.yaml
+
+    #TODO add timeout
+    pod_status=$(kubectl get pods | grep mofed-test-pod-1 |awk  '{print $3}')
+    while [ $pod_status != 'Running' ]; do
+        echo "Waiting for pod to became Running"
+        pod_status=$(kubectl get pods | grep mofed-test-pod-1 |awk  '{print $3}')
+        sleep 30
+    done
+    kubectl describe pod mofed-test-pod-1
+    return $?
+}
+
+
+function test_pod {
+    kubectl exec -i mofed-test-pod-1 -- ip a
+    return $?
+}
+
+
+function clear_vfs {
+    echo 0 > /sys/class/net/$SRIOV_INTERFACE/device/sriov_numvfs
+    echo 4 > /sys/class/net/$SRIOV_INTERFACE/device/sriov_numvfs
+}
+
+
+function install_k8s {
+    echo "Stopping previous K8S run"
+    #TODO clean on exit
+    kill $(ps -ef |grep kube |awk '{print $2}')
+    go get -d k8s.io/kubernetes
+    cd $GOPATH/src/k8s.io/kubernetes
+    git log -p -1 > $ARTIFACTS/kubernetes.txt
+    make
+    go get -u github.com/tools/godep
+    go get -u github.com/cloudflare/cfssl/cmd/...
+    $GOPATH/src/k8s.io/kubernetes/hack/install-etcd.sh
+    export PATH="$GOPATH/src/k8s.io/kubernetes/third_party/etcd:${PATH}"
```

Comment:
`export PATH="$GOPATH/src/k8s.io/kubernetes/third_party/etcd:${PATH}"`
This is needed always to run k8s, consider moving it to the export PATH at the start of the script